lib/Runtime/PlatformAgnostic/Platform/Common/NumbersUtility.cpp

```diff
@@ -0,0 +1,235 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "RuntimePlatformAgnosticPch.h"
+#include "Common.h"
+#include "ChakraPlatform.h"
+#include <locale.h>
+
+namespace PlatformAgnostic
+{
+namespace Numbers
+{
+    static const WCHAR number_zero        = WCHAR('0');
+    static const WCHAR number_five        = WCHAR('5');
+    static const WCHAR number_nine        = WCHAR('9');
+
+    Utility::NumbersLocale::NumbersLocale()
+    {
+        maxDigitsAfterDecimals = 3;
+        defaultDecimalDot = WCHAR('.');
+        defaultDecimalComma = WCHAR(',');
+
+        // get process/LC_NUMERIC
+        char *old_locale = setlocale(LC_NUMERIC, NULL);
+
+        // get user's LC_NUMERIC from OS
+        setlocale(LC_NUMERIC, "");
+        struct lconv *locale_format = localeconv();
+        localeThousands = (WCHAR) *(locale_format->thousands_sep);
+        localeDecimal = (WCHAR) *(locale_format->decimal_point);
+        localeNegativeSign = (WCHAR) *(locale_format->negative_sign);
+
+        // set process/LC_NUMERIC back
+        setlocale(LC_NUMERIC, old_locale);
+    }
+
+    Utility::NumbersLocale Utility::numbersLocale;
+
+    size_t Utility::NumberToDefaultLocaleString(const WCHAR *number_string,
```

Comment:
I don't think this covers all the cases the GetNumberFormatEx would support.
(e.g. I can switch the standard digits character to Chinese characters).

My guess is that ICU have support for richer locale specific number formatting as well?
For now, this custom implementation probably is sufficient for the case ICU support is not included.
But for windows, we should stick to GetNumberFormatEx so that we don't regress functionality.